File: 080_remove_duplicates_from_sorted_array_ii/Solution.py
class Solution:
    def removeDuplicates(self, nums: list) -> int:
        arr_len = len(nums)
        if arr_len <= 2: return arr_len

        i, count = 0, 1
        for j in range(1, arr_len):
            if nums[i] == nums[j]: count += 1
            else: count = 1

            if count <= 2:
                i += 1
                nums[i] = nums[j]

        return i + 1


if __name__ == '__main__':
    nums = [1, 1, 1, 2, 2, 3]
    print(Solution().removeDuplicates(nums))
    print(nums)

File: 080_remove_duplicates_from_sorted_array_ii/Solution.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length <= 2) return nums.length;

        int count = 1, j = 0;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[j]) count++;
            else count = 1;

            if (count <= 2) nums[++j] = nums[i];
        }

        return j + 1;
    }

    public static void main(String[] args) {
        int[] nums = {1, 2, 2};
        System.out.println(new Solution().removeDuplicates(nums));
        for (int num : nums)
            System.out.print(num + " ");
    }
}
